index.js

```diff
@@ -1,24 +1,32 @@
 'use strict';
 
-var minimatch = require('minimatch');
+var micromatch = require('micromatch');
+var arrify = require('arrify');
 
 var anymatch = function(criteria, value, returnIndex, startIndex, endIndex) {
-  if (!Array.isArray(criteria)) { criteria = [criteria]; }
-  if (arguments.length === 1) { return anymatch.bind(null, criteria); }
-  var string = Array.isArray(value) ? value[0] : value;
-  if (!startIndex) { startIndex = 0; }
+  criteria = arrify(criteria);
+  value = arrify(value);
+  if (arguments.length === 1) {
+    return criteria.length === 1 ?
+      micromatch.matcher(criteria[0]) :
+      anymatch.bind(null, criteria.map(function(criterion) {
```

Comment:
@jonschlinkert btw have you considered adding array support to `micromatch.matcher`? It would make it possible to simplify this bit.